### PR TITLE
feat(Checkbox): improve accessibility, tests and use in docs

### PR DESCRIPTION
**Pull request by jurokapsiar**
Adds aria-disabled
Adds accessibility handling and conformance unit tests
Replaces the use of input checkbox with the Checkbox component in the docs

### CODE REVIEW

**Review by sophieH29 — APPROVED** [review]

File: packages/react/src/lib/accessibility/Behaviors/Checkbox/checkboxBehavior.ts [hunk 1]
```diff
@@ -3,26 +3,29 @@ import { Accessibility } from '../../types'
...
  * Adds role='checkbox'. This allows screen readers to handle the component as a checkbox button.
  * Adds attribute 'aria-checked=true' based on the property 'checked'.
+ * Adds attribute 'aria-disabled=true' based on the property 'disabled'.
  * Adds attribute 'tabIndex=0' to 'root' component's part.
  */
 const checkboxBehavior: Accessibility<CheckboxBehaviorProps> = props => ({
   attributes: {
     root: {
       'aria-checked': props.checked,
+      'aria-disabled': props.disabled,
```
Comment (miroslavstastny) [inline packages/react/src/lib/accessibility/Behaviors/Checkbox/checkboxBehavior.ts:20]: I am wondering why we have different logic implemented here vs `buttonBehavior` vs `menuItemBehavior`. Should we use just this simple mapping or anything else and reuse it everywhere?

File: docs/src/components/ComponentDoc/ComponentProps/ComponentProps.tsx [hunk 2]
```diff
@@ -54,27 +54,14 @@ export default class ComponentProps extends React.Component<any, any> {
...
-            <Flex.Item>
-              <ComponentPropsComponents
-                activeDisplayName={activeDisplayName}
-                displayNames={displayNames}
-                onItemClick={this.handleComponentClick}
-                parentDisplayName={displayName}
-              />
-            </Flex.Item>
+          <Flex gap="gap.medium">
+            <Checkbox checked={!!activeDisplayName} onClick={this.handleToggle} label="Props" />
```
Comment (layershifter) [inline docs/src/components/ComponentDoc/ComponentProps/ComponentProps.tsx]:
```suggestion
            <Checkbox checked={!!activeDisplayName} onChange={this.handleToggle} label="Props" />
```

File: packages/react/src/lib/accessibility/Behaviors/Checkbox/checkboxBehavior.ts (hunk 1 above)
Comment (jurokapsiar) [reply packages/react/src/lib/accessibility/Behaviors/Checkbox/checkboxBehavior.ts:20]: yes, this is overcomplicated in the case of button and menu item. we will simplify those.

File: packages/react/src/lib/accessibility/Behaviors/Checkbox/checkboxBehavior.ts [hunk 3]
```diff
@@ -3,26 +3,29 @@ import { Accessibility } from '../../types'
 
 type CheckboxBehaviorProps = {
   checked: boolean
+  disabled: boolean
```
Comment (sophieH29) [inline packages/react/src/lib/accessibility/Behaviors/Checkbox/checkboxBehavior.ts]: shouldn't be disabled an optional prop? If it is not passed then we don't render `aria-diabled`

**Pull request merged**